week7/answers.sql

```diff
@@ -1,3 +1,314 @@
 -- Create your tables, views, functions and procedures here!
 CREATE SCHEMA destruction;
 USE destruction;
+
+CREATE TABLE players (
+	player_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	first_name VARCHAR(30) NOT NULL,
+	last_name VARCHAR(30) NOT NULL,
+	email VARCHAR(50) NOT NULL
+);
+  
+CREATE TABLE characters (
+	character_id INT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
+	player_id INT UNSIGNED,
+	name VARCHAR(30) NOT NULL,
+	level TINYINT,
+
+	CONSTRAINT characters_fk_players
+		FOREIGN KEY (player_id)
+		REFERENCES players (player_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE
+);
+
+CREATE TABLE winners (
+	character_id INT UNSIGNED,
+	name VARCHAR(30) NOT NULL,
+
+	CONSTRAINT winners_fk_characters
+		FOREIGN KEY (character_id)
+		REFERENCES characters (character_id)
+		ON UPDATE CASCADE
+		ON DELETE CASCADE
+);
+
+CREATE TABLE character_stats (
+	character_id INT UNSIGNED,
```

Comment:
This should also be declared as a PRIMARY KEY column. It is okay for a PRIMARY KEY to also be a FOREIGN KEY. This will prevent the possibility of two of the same characters being inserted and satisfies editor constraints for updating/deleting rows in MySQL Workbench (when we don't have the safety turned off, like we do).